.circle/Makefile

```diff
@@ -69,6 +67,8 @@ compile:
 	@echo "======================= compile ========================"
 	@echo "------- Compile all .py files (syntax check test) ------"
 	if python -c 'import compileall,re; compileall.compile_dir(".", rx=re.compile(r"/virtualenv|virtualenv-osx|virtualenv-py3|.tox|.git|.venv-st2devbox"), quiet=True)' | grep .; then exit 1; else exit 0; fi
+	@echo
+	@echo "All files compiled successfully"
```

Comment:
There is exit 0 above and the command doesn't run in a subshell so I assume this output will never be printed, but I could be wrong.

Reply:
Oh, I think you're right.

Reply:
Nope, and I figured out why. Each line of a `Makefile` is run in its own Bash shell, that's why exporting variables across lines of a make recipe don't work, or activating a virtualenv in one line and using it in another.

So if the `if python -c` line fails, `make` will catch it and stop. If it doesn't, then that line exits with status `0`, and the echo lines are run in their own Bash shells.